oprf/oprf_test.go

```diff
@@ -0,0 +1,387 @@
+		if err != nil {
+			t.Fatal("invalid setup of server: " + err.Error())
+		}
+		if srv == nil {
+			t.Fatal("invalid setup of server: no server.")
+		}
+
+		privKey, _ := hex.DecodeString(v.PrivK)
+		srv.Kp.PrivK.Set(privKey)
+
+		client, err := NewClient(OPRFP384)
+		if err != nil {
+			t.Fatal("invalid setup of client: " + err.Error())
+		}
+
+		token, bToken := blindTest(client.suite, j)
+		testBToken, _ := hex.DecodeString(j.Blind.Blinded)
+		if !bytes.Equal(testBToken[:], bToken[:]) {
+			test.ReportError(t, bToken[:], testBToken[:], "request")
+		}
+
+		eval, _ := srv.Evaluate(bToken)
+		if eval == nil {
+			t.Fatal("invalid evaluation of server: no evaluation.")
+		}
+
+		testEval, _ := hex.DecodeString(j.Evaluation.Eval)
+		if !bytes.Equal(testEval[:], eval.element[:]) {
+			test.ReportError(t, eval.element[:], testEval[:], "eval")
+		}
+
+		info := []byte("test information")
+		iToken, h, _ := client.Finalize(token, eval, info)
+
+		testIToken, _ := hex.DecodeString(j.Unblind.IToken)
+		if !bytes.Equal(testIToken[:], iToken[:]) {
+			test.ReportError(t, iToken[:], testIToken[:], "finalize")
+		}
+
+		testOutput, _ := hex.DecodeString(j.Output)
+		if !bytes.Equal(testOutput[:], h[:]) {
+			test.ReportError(t, h[:], testOutput[:], "finalize")
+		}
+	}
+}
+
+func (v *Vectors) runP521(t *testing.T) {
+	for _, j := range v.Vector {
+		srv, err := NewServer(OPRFP521, nil, nil)
+		if err != nil {
+			t.Fatal("invalid setup of server: " + err.Error())
+		}
+		if srv == nil {
+			t.Fatal("invalid setup of server: no server.")
+		}
+
+		privKey, _ := hex.DecodeString(v.PrivK)
+		srv.Kp.PrivK.Set(privKey)
+
+		client, err := NewClient(OPRFP521)
+		if err != nil {
+			t.Fatal("invalid setup of client: " + err.Error())
+		}
+
+		token, bToken := blindTest(client.suite, j)
+		testBToken, _ := hex.DecodeString(j.Blind.Blinded)
+		if !bytes.Equal(testBToken[:], bToken[:]) {
+			test.ReportError(t, bToken[:], testBToken[:], "request")
+		}
+
+		eval, _ := srv.Evaluate(bToken)
+		if eval == nil {
+			t.Fatal("invalid evaluation of server: no evaluation.")
+		}
+
+		testEval, _ := hex.DecodeString(j.Evaluation.Eval)
+
+		if !bytes.Equal(testEval[:], eval.element[:]) {
+			test.ReportError(t, eval.element[:], testEval[:], "eval")
+		}
+
+		info := []byte("test information")
+		iToken, h, _ := client.Finalize(token, eval, info)
+
+		testIToken, _ := hex.DecodeString(j.Unblind.IToken)
+		if !bytes.Equal(testIToken[:], iToken[:]) {
+			test.ReportError(t, iToken[:], testIToken[:], "finalize")
+		}
+
+		testOutput, _ := hex.DecodeString(j.Output)
+		if !bytes.Equal(testOutput[:], h[:]) {
+			test.ReportError(t, h[:], testOutput[:], "finalize")
+		}
+	}
+}
+
+func TestDraftVectors(t *testing.T) {
+	// Test vectors from draft-05
+	var v Vectors
+
+	v.readFile(t, "testdata/256_vectors.json")
+	t.Run("ORPF-P256", v.runP256)
+
+	v.readFile(t, "testdata/384_vectors.json")
+	t.Run("ORPF-P384", v.runP384)
+
+	//v.readFile(t, "testdata/521_vectors.json")
```

Comment:
Why is this commented out?

Reply:
I found a bug on 521 last night. So I have to check those vectors.

Reply:
Okay, thanks!

Reply:
Fixed now.